auto_install/install.sh

```diff
@@ -585,33 +585,35 @@ preconfigurePackages(){
 	fi
 
 	if [[ "${PLAT}" == 'Alpine' ]] && ! command -v grepcidr &> /dev/null; then
+		local down_dir
 		## install dependencies
 		# shellcheck disable=SC2086
 		${SUDO} ${PKG_INSTALL} build-base make curl tar
 
-		## download binaeries
-		curl -fLo master.tar.gz https://github.com/pivpn/grepcidr/archive/master.tar.gz
-		tar -xzf master.tar.gz
-
-		cd grepcidr-master || exit 1
+		if ! down_dir="$(mktemp -d)"; then
+			echo "::: Failed to create download directory for grepcidr!"
+			exit 1
+		fi
 
-		## personalize binaries
-		sed -i -E -e 's/^PREFIX\=.*/PREFIX\=\/usr\nCC\=gcc/' Makefile
+		## download binaries
+		curl -fLo "${down_dir}/master.tar.gz" https://github.com/pivpn/grepcidr/archive/master.tar.gz
+		tar -xzC "${down_dir}" -f "${down_dir}/master.tar.gz"
 
-		## install
-		make
-		${SUDO} make install
+		(
+			cd "${down_dir}/grepcidr-master" || exit 1
 
-		if ! command -v grepcidr &> /dev/null; then
-			echo "::: Failed to compile and install grepcidr!"
-			exit 1
-		fi
+			## personalize binaries
+			sed -i -E -e 's/^PREFIX\=.*/PREFIX\=\/usr\nCC\=gcc/' Makefile
 
-		cd ..
+			## install
+			make
+			${SUDO} make install
 
-		## remove useless files
-		rm master.tar.gz
-		rm -rf grepcidr-master
+			if ! command -v grepcidr &> /dev/null; then
+				echo "::: Failed to compile and install grepcidr!"
+				exit 1
+			fi
+		) || exit 1
```

Comment:
Why wrapping with `(...) || exit 1`? Seems unnecessary considering the inner `exit 1`.

Reply:
> The inner exit exits only from the subshell

It does that because it’s wrapped in (…), but it did exit the script before.

Reply:
Yes, but now I don't have to `cd` a subdirectory of my current directory anymore
I have to enter into a directory with a different path and through this trick I avoid using specific Bash functions (in case one day we want to switch completely to the POSIX Shell) and saving the path of the directory I am in and then returning to it (see the piece of code below)
```
local base_dir

base_dir="${PWD}"

cd "${down_dir}/grepcidr-master" || exit 1

...

cd "${base_dir}" || exit 1
```